### PR TITLE
Add sync batchnorm

### PR DESCRIPTION
**Pull request by DTennant**
add sync batchnorm(implementation from this [repo](https://github.com/vacancy/Synchronized-BatchNorm-PyTorch)) for the multi-gpu training, also apply linear scale rule from https://arxiv.org/abs/1706.02677

### CODE REVIEW

**Comment by L1aoXingyu** [comment]
Please do some performance tests on reid datasets, such as market1501 rank1 etc. and speed test. We want to get the same result for rank1 and mAP for syncBN when batch_size=64.
PS: Please put the sync_bn in `layers` folder to make our code base structured.

**Comment by DTennant** [comment]
Sorry for not replying for so long, I have some other thing to work on..
Here is the mAP and CMC result for syncbn with batchsize 64, I was using two GTX 1080Ti

```
2019-05-09 16:12:22,187 reid_baseline.train INFO: Epoch 120 done. Time per batch: 83.640[s] Speed: 140.8[samples/s]
2019-05-09 16:12:22,187 reid_baseline.train INFO: ----------
The test feature is normalized
2019-05-09 16:14:43,836 reid_baseline.train INFO: Validation Results - Epoch: 120
2019-05-09 16:14:43,836 reid_baseline.train INFO: mAP: 85.9%
2019-05-09 16:14:43,836 reid_baseline.train INFO: CMC curve, Rank-1  :94.1%
2019-05-09 16:14:43,836 reid_baseline.train INFO: CMC curve, Rank-5  :98.2%
2019-05-09 16:14:43,837 reid_baseline.train INFO: CMC curve, Rank-10 :99.0%
```

I was using the config file [here](https://github.com/DTennant/reid-strong-baseline/blob/master/configs/softmax_triplet_with_center_multi_gpu.yml)
